event-handling/adding-onclick-handling.md

```diff
@@ -0,0 +1,173 @@
+
+### !callout-info
+
+## Why Do We Listen for DOMContentLoaded?
+
+We need to ask our document to wait and register the event handlers _after_ the DOM has been completely loaded. This is because the JavaScript file often loads faster than the DOM does! If we don't wait for the DOM to load, then our JS code will not find our `crabButton` in the DOM, and `crabButton` will be `null`. It's important to consider what's on the DOM whenever we search for an element.
+
+<br/>
+
+In other libraries, this problem may already be solved by the library, without us registering a DOMContentLoaded handler.
+
+### !end-callout
+
+### Verifying
+
+Let's check our website now to see if we can add some crabs!
+
+![Crab Fan Site, with seven crabs added to the bottom as a result of clicking the "Add Crab" button seven times.](../assets/event-handling_adding-onclick-handling.png)
+
+Ideally, we should see that clicking the "Add Crab" button adds a crab to the page.
+
+If we don't have this result, different ways we can debug include:
+
+- Checking the HTML and JS to make sure there is:
+  - A valid "Add Crab" button that can be found
+  - A valid "Crab" section that can be found
+- The names of events are case-sensitive
+
+### The Event Object
+
+When we introduced the `addEventListener` function, we mentioned that the _event-handling_ _callback_ function we passed in could expect an _event_ argument.
+
+"Events" _in themselves_ are objects. Event objects contain all the information about the event, such as what type of event it is, and where it happened. The following events would all separate event objects:
+
+1. A click event that happens on the `addCrab` button at 12:00
+1. A click event that happens on the `addCrab` button at 12:01
+1. A click event that happens on the `addCrab` button at 12:02, and so on.
+
+By default, our browser will pass in the event object (the object that represents the event) into every event-handler function.
+
+Access to the Event object can give us extra details about the event, and can help us customize our event-handling.
+
+[Let's change our JS code](https://replit.com/@adacore/Adding-onclick-Handling-Demo#scripts/index.js) to anticipate and access the event object that comes into every event-handler. Notice that we are only changing two things:
+
+- The parameters for event-handling functions
+- Accessing the events inside the functions in print statements
+
+```js
+const addCrab = (event) => {
+  console.log("in addCrab:", event);
+  // ... other addCrab functionality
+};
+
+const registerEventHandlers = (event) => {
+  console.log("in registerEventHandlers:", event);
+  // ... other registerEventHandler functionality
+};
+
+// ...
+```
+
+When we run this, we can see in the console that we get details about the "DOMContentLoaded" event when the page loads. Also, every time we add a crab, we get details about the "click" event (helpfully labeled as `MouseEvent` in the console).
```

Comment:
'also,' feels like it brings everything to a temporary halt. keep it flowing
```suggestion
When we run this, we can see in the console that we get details about the "DOMContentLoaded" event when the page loads. And every time we add a crab, we get details about the "click" event (helpfully labeled as `MouseEvent` in the console).
```

Reply:
The Event object is worth exploring, but in this context, I worry that it would just be a "here's a bag of stuff!" without a lot of grounding. I'd be a little more inclined to only explicitly introduce the parts of Event they need as they need them, and the rest to the reader's curiosity.

Reply:
I like the idea of exploring an Event object, but I think that's a tangent that can be explored during this time, during React, or... uh never lol. To Ansel's point, it'd be nice if students had a more concrete objective for exploring it.

Another reason why I'll punt the Event object exploration in text atm is because documentation for Event objects is pretty terrible imo 🙀 aaaaand I'll just sidestep that problem for now by not including it :)